system-addon/content-src/components/Base/Base.jsx

```diff
@@ -46,11 +54,18 @@ class Base extends React.Component {
     const props = this.props;
     const {locale, strings, initialized} = props.App;
     const prefs = props.Prefs.values;
-    if (!initialized || !strings) {
+    if (!strings) {
+      return null;
+    }
+
+    if (!props.isPrerendered && !initialized) {
       return null;
     }
 
-    return (<IntlProvider key={locale} locale={locale} messages={strings}>
+    // Note: the key on IntlProvider must be static in order to not blow away
+    // all elements on a locale change (such as after preloading).
+    // See https://github.com/yahoo/react-intl/issues/695 for more info.
+    return (<IntlProvider key="STATIC" locale={locale} messages={strings}>
```

Comment:
This change seems to cause the search placeholder and migration texts to not appear on the first about:home. (You can also test with `./mach run about:newtab`) Changing back to key={locale} avoids this:
<img width="771" alt="image" src="https://user-images.githubusercontent.com/438537/30137292-7e279a0e-9317-11e7-9cc7-2b196b531575.png">

My guess is that `LOCALE_UPDATED` happens at an unexpected time?

Reply:
Looks like passing in a dummy `locale` prop, e.g., `<Search locale={locale} />`, makes it `render` again